models/fates/R/write.configs.FATES.R

```diff
@@ -20,133 +20,156 @@
-  Sys.chmod(file.path(settings$rundir, run.id, "job.sh"))
-  # ## Write PARAMETER file ## Write SETTINGS file
-} # write.config.FATES
+
+write.config.FATES <- function(defaults, trait.values, settings, run.id){
+   library(PEcAn.utils)
+#  
+# #OUTLINE OF MODULES
+#   # Copy Case and Build
+#   #  -symbolic link to refernce case that is already completed
+#   # Edit user_nl_* files to add site info
+#   # make Jobs.sh -case_submit
+#   # call met2model and add to namelists
+#   #
+
+   site <- settings$run$site
+   site.id <- as.numeric(site$id)
+  
+   # find out where things are
+   local.rundir <- file.path(settings$rundir, run.id) ## this is on local machine for staging
+   rundir <- file.path(settings$host$rundir, run.id)  ## this is on remote machine for execution
+   casedir <- file.path(rundir,"case") 
+   outdir <- file.path(settings$host$outdir, run.id)
+   refcase   <- settings$model$binary
+   bld    <- file.path(refcase,"bld")
+   binary <- file.path(bld,"cesm.exe")
+   indir  <- file.path(rundir,"input") ## input directory
+   default <- settings$run$inputs$default$path ## reference inputs file structure
+   site.id
+   site_name <- paste0(site.id %/% 1000000000, "-", site.id %% 1000000000)
+   
+   ## DATES
+   ## CLM is a bit odd and takes a start date and length, so we need to precompute
+   ## this needs to be generalized to fractional years, but accounting for 365 day year
+   start_date <- as.Date(settings$run$start.date)
+   end_date   <- as.Date(settings$run$end.date)
+   stop_n     <- as.numeric(end_date - start_date, units="days") - n_leap_day(start_date,end_date) + 1  
+   
+   
+   ##-----------------------------------------------------------------------##
+   ##                                                                       ##
+   ##                             INPUTS                                    ##
+   ##                                                                       ##
+   ##-----------------------------------------------------------------------##
+
+   ## SITE INFO --> DOMAIN FILE (lat/lon)
+   gridres = 0.125  ## ultimately this should be a variable
+   lat = site$lat
+   lon = (site$lon + 360) %% 360 ## make sure coords in 0-360 range, not negative
+   domain.default <- system.file("domain.lnd.1x1pt-brazil_navy.090715.nc",package="PEcAn.FATES")
+   domain.file <- file.path(local.rundir,paste0("domain.lnd.",site_name,".nc"))
+   file.copy(domain.default,domain.file)
+   domain.nc <- nc_open(domain.file,write=TRUE)
+   ncvar_put(nc=domain.nc, varid='xc', vals=lon)
+   ncvar_put(nc=domain.nc, varid='yc', vals=lat)
+   ncvar_put(nc=domain.nc, varid='xv', vals=lon+c(-1,1,1,-1)*gridres)
+   ncvar_put(nc=domain.nc, varid='yv', vals=lat+c(-1,-1,1,1)*gridres)
+   ncvar_put(nc=domain.nc, varid='area', vals=(2*gridres*pi/180)^2)   
+   nc_close(domain.nc)
+   
+   ## SURF
+   surf.default <- "/home/carya/FATESinput/lnd/clm2/surfdata_map/surfdata_1x1_brazil_16pfts_simyr2000_c160127.nc"
```

Comment:
@serbinsh all this is still in development and this was the quickest hack to get started.

That said, system.file is an option, but I think it'd be better to grab this from the reference case so that CLM-FATES users can update that reference case that are more site- or system-specific (this file includes things like soils, etc).